docs/howto/how_to_debug.md

```diff
@@ -0,0 +1,162 @@
+# How to Debug Apollo
+
+## Debugging Apollo
+
+The Apollo project runs in Docker and cannot be used directly on the host machine. It must be created in Docker with GDBServer. Debug the service process, and then use GDB to connect to the debug service process in Docker on the host machine. The specific operation methods are as follows:
+
+### Prerequisites
+The main prerequisites contain collecting debugging information and installing the GDBServer if it is not already present in Docker
+
+#### Collecting debugging information
+
+When compiling Apollo projects, you will need to use debugging information options such as **build_dbg** or **build_gpu**. Optimization options such as **build_opt** or **build_opt_gpu** cannot be used.
+
+#### Install GDBServer inside Docker
+
+After entering Docker, you can use the following command to view if the GDBServer is present:
+
+``` bash
+gdbserver --version
+```
+
+If the prompt is similar to the following information:
+
+```
+GNU gdbserver (Ubuntu 7.7.1-0ubuntu5~14.04.3) 7.7.1
```

Comment:
GNU gdbserver (Ubuntu 8.1-0ubuntu3.2) 8.1.0.20180409-git
Copyright (C) 2018 Free Software Foundation, Inc.
gdbserver is free software, covered by the GNU General Public License.
This gdbserver was configured as "x86_64-linux-gnu"